iceoryx_utils/source/posix_wrapper/timer.cpp

```diff
@@ -208,6 +254,10 @@ Timer::OsTimer::~OsTimer() noexcept
     {
         stop();
 
+        // do not delete the timer while the callback is running, it could access the timer which is about to be deleted
```

Comment:
Should address the segfault in the destructor, at least it could not be reproduced any more and it definitely was a problem, but there may be additional causes.